### PR TITLE
Improvement/member_resolves

### PR DESCRIPTION
**Pull request by worksofliam**
### Changes

This adds an additional member resolve API, which will eventually deprecate the previous `memberResolves` API. This API depends on a brand new CL, which will inhherit the user profile iASP, which the current API does not.

I also created tests for this new API.

### Checklist

* [x] have tested my change
* [ ] updated relevant documentation
* [ ] Remove any/all `console.log`s I added
* [ ] eslint is not complaining
* [ ] have added myself to the contributors' list in [CONTRIBUTING.md](https://github.com/halcyon-tech/vscode-ibmi/blob/master/CONTRIBUTING.md)
* [ ] **for feature PRs**: PR only includes one feature enhancement.

### CODE REVIEW

**Comment by worksofliam** [comment]
Converted to draft as there may be an easier way with our existing API after all....

File: src/api/IBMiContent.ts [hunk 1]
```diff
@@ -649,6 +651,32 @@ export default class IBMiContent {
     return items;
   }
 
+  async memberResolveLibraryList(member: string, file: string, libraryList?: ILELibrarySettings): Promise<IBMiMember | undefined> {
+    if (this.ibmi.remoteFeatures[`GETMBR.PGM`]) {
+      let env: {[id: string]: string} = {};
+
+      if (libraryList) {
+        if (libraryList.currentLibrary) env[`&CURLIB`] = libraryList.currentLibrary;
+        if (libraryList.currentLibrary) env[`&LIBL`] = libraryList.libraryList.join(` `);
```
Comment (chrjorgensen) [inline src/api/IBMiContent.ts:660]: Shouldn't this be `libraryList.libraryList`?

```suggestion
        if (libraryList.libraryList) env[`&LIBL`] = libraryList.libraryList.join(` `);
```

File: src/languages/clle/getmbr.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    const tempLib = config.tempLibrary;
+  
+    try {
+      await connection.remoteCommand(`CRTSRCPF ${tempLib}/QTOOLS`, undefined)
+    } catch (e) {
+      //It may exist already so we just ignore the error
+    }
+  
+    await content.uploadMemberContent(undefined, tempLib, `QTOOLS`, `GETMBR`, getSource());
+    await connection.remoteCommand(
```
Comment (chrjorgensen) [inline src/languages/clle/getmbr.ts:21]: `remoteCommand` is deprecated and `runCommand` should be used instead:

```suggestion
    await connection.runCommand(
```

**Review by chrjorgensen — CHANGES_REQUESTED** [review]
@worksofliam I like this... Only a few minor issues... 😃

**Comment by worksofliam** [comment]
I have thought of a better way to do this PR so I am going to close and have another branch for it in the future.

**Pull request closed without merging**